### PR TITLE
Stop flow on teamcity

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

To quote @SiAdcock from Trello https://trello.com/c/Hns5LbUf/438-flow-state-issue-on-teamcity-agents

> When Flow runs on a TeamCity agent, the Flow server is not stopped. As a result, the next time Flow runs on this agent, even if the branch has changed, Flow validates the code in the new branch against the state of the file system of the old branch. Therefore any new modules added in the new branch will not exist from Flow's perspective.

This change stops Flow server, meaning the next time Flow runs on the Teamcity agent it will restart the server afresh.

## What is the value of this and can you measure success?

Prevent intermittent flow issue on Teamcity that results in a red build.

## Does this affect other platforms - Amp, Apps, etc?

No

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

No

## Screenshots

N/A

## Tested in CODE?
No

### CODE REVIEW

**Comment by sndrs** [comment]
i wonder if this is in part due to multiple flow configs? maybe we should try and stop flow before running in validation itself?

**Comment by SiAdcock** [comment]
It will be difficult to know for sure, unless we wait for the problem to happen again, and then try building this change on the problematic agent. 

It's worth noting that regardless of whether this is the correct fix for this problem, guardian/support-frontend#201 introduced a similar change to this, to fix a different issue. It might be worth talking to @svillafe to see if he can confirm exactly that the problem was... He did say that his change fixed it though!

**Comment by GHaberis** [comment]
@sndrs do you mean stop flow before running `./tools/task-runner/runner validate test --teamcity`...? I can move the new commands before this if so and see what happens on TC.

**Comment by GHaberis** [comment]
@sndrs I've moved the `flow stop` command before validation and everything has ran OK in Teamcity.

File: dev/teamcity/dist-assets-tc [hunk 1]
```diff
@@ -11,6 +11,24 @@ set -x
 
 echo "Asset compilation"
 
+cd static/src/javascripts
+
+set +x
```
Comment (nicl) [inline dev/teamcity/dist-assets-tc]: I see that this is done throughout this file and not unique to this PR, but do you know why we wrap most commands in `+x` and `-x`? This toggles the debug flag but I don't understand why we would, for example, wrap a simple echo command with this.

Comment (sndrs) [reply dev/teamcity/dist-assets-tc]: i think `+x` toggles it off? then `set -x` toggles it back on after echoing?

Comment (GHaberis) [reply dev/teamcity/dist-assets-tc]: Any commands between `set +x` and `set -x` are printed out to the terminal, which has it's uses, but I agree with @nicl it seems redundant for `echo` commands which should be printed out anyway.

Comment (nicl) [reply dev/teamcity/dist-assets-tc]: Yeah, if it is really to toggle things off with echo then a custom function is probably better than this repetition all over the place.

**Comment by NataliaLKB** [comment]
@GHaberis what is the status of this PR?

**Comment by GHaberis** [comment]
Closing this PR as the issue hasn't occurred in the past few weeks, there have been other changes to our `flow` setup that may have fixed this issue. If it occurs again we can resurrect this.

**Pull request closed without merging**